nginx-controller/nginx/configurator.go

```diff
@@ -766,6 +781,49 @@ func (cnf *Configurator) updatePlusEndpoints(ingEx *IngressEx) error {
 	return nil
 }
 
+func filterMasterAnnotations(annotations map[string]string) []string {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range masterBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return removedAnnotations
+}
+
+func filterMinionAnnotations(annotations map[string]string) []string {
+	var removedAnnotations []string
+
+	for _, blacklistAnn := range minionBlacklist {
+		if _, ok := annotations[blacklistAnn]; ok {
+			removedAnnotations = append(removedAnnotations, blacklistAnn)
+			delete(annotations, blacklistAnn)
+		}
+	}
+
+	return removedAnnotations
+}
+
+func mergeMasterAnnotationsIntoMinion(minionAnnotations map[string]string, masterAnnotations map[string]string) {
+	var isBlacklisted bool
+
+	for key, val := range masterAnnotations {
+		if _, ok := minionAnnotations[key]; !ok {
+			for _, blacklistAnn := range minionBlacklist {
```

Comment:
the check if an annotation is in the black list has a bug -- `isBlacklisted` must be initialized to `false` prior to each iteration of the loop.

Instead of iterating over the minionBlacklist, I suggest to change the minionBlacklist from `[]string` to `map[string]bool` for performance reason. The code will look like this:
```golang
	for key, val := range masterAnnotations {
		if _, ok := minionAnnotations[key]; !ok && !minionBlacklist[key] {
			minionAnnotations[key] = val
		}
	}
```

Reply:
I can change to maps in another PR if that ok.

Reply:
sounds good!